File: miniJobs/Domain/Entities/City.cs
using Domain.Interfaces;

namespace Domain.Entities;

[Table("cities")]
public class City : IEntity<int>
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("name")]
    public string Name { get; set; }

    [Column("postcode")]
    public string Postcode { get; set; }

    [Column("municipality_code")]
    public string MunicipalityCode { get; set; }

    [Column("country_id")]
    public int? CountryId { get; set; }

    [NotMapped]
    public Country Country { get; set; }

    [Column("is_deleted")]
    public bool IsDeleted { get; set; }
}

File: miniJobs/UI/minijobs_admin/lib/models/city.dart
import 'package:json_annotation/json_annotation.dart';

part 'city.g.dart';

@JsonSerializable()
class City {
  int? id;
  String? name;
  int? countryId;
  String? municipalityCode;
  String? postcode;
  bool? isDeleted;
  String? countryName;


  City(this.id, this.name, this.countryId, this.municipalityCode,
      this.postcode,this.isDeleted,this.countryName);

  factory City.fromJson(Map<String, dynamic> json) => _$CityFromJson(json);

  Map<String, dynamic> toJson() => _$CityToJson(this);
}
